pageserver/src/tenant/timeline.rs

```diff
@@ -158,7 +161,7 @@ pub struct Timeline {
 
     // List of child timelines and their branch points. This is needed to avoid
     // garbage collecting data that is still needed by the child timelines.
-    pub gc_info: RwLock<GcInfo>,
+    pub gc_info: tokio::sync::RwLock<GcInfo>,
```

Comment:
Why this change? This lock protects just a few variables..

Hmm, I can see the held while making pretty complicated calls in `update_gc_info`. `update_gc_info` calls `find_lsn_for_timestamp` in particular. I think we should refactor `update_gc_info` so that it would do the `find_lsn_for_timestamp` call before acquiring this lock.

(not a blocker, I'm ok to punt that to a followup PR)

Reply:
Good find, it's not really necessary. Will amend the original preliminary refactor commit that introduced it in the next force push.